src/kepler_model/train/extractor/extractor.py

```diff
@@ -229,15 +236,20 @@ def get_system_feature_data(self, query_results, features):
         return feature_data
 
     # return with timestamp index
-    def get_power_data(self, query_results, energy_components, source):
+    def get_power_data(self, query_results, energy_components, source, use_vm_metrics=False):
         power_data_list = []
         for component in energy_components:
             unit_col = get_energy_unit(component)  # such as package
-            query = energy_component_to_query(component)
+            if use_vm_metrics:
+                query = vm_energy_component_to_query(component)
+            else:
+                query = energy_component_to_query(component)
             if query not in query_results:
                 print(query, "not in", query_results)
                 return None
             aggr_query_data = query_results[query].copy()
+            if not use_vm_metrics:
```

Comment:
Could you please comment an explanation of this line?

Reply:
So if it is baremetal, I do not want it to use job=vm energy metric queries. We should only use baremetal metrics. Basically, I am making sure that the behavior of the extractor prior to my addition of vm metrics is the same as before (which is it will extract all the metrics with kepler_node_*). Same with in get_workload_feature_dataset.

Reply:
can we have a single splitter to split between vm_query_results and bm_query_results from a single query_result? It could be apply for the feature extract as well.  
Here I think it is fine to filter only for bm query since the vm query has `vm` word which is only available on bm. However, as I mentioned below, I still doubt to use vm value not the node value.

Reply:
Good point. I will change that now

Reply:
Can we merge this as is because the issue is a lot of the methods in DefaultExtractor which imo should be protected are being used directly (like get_workload_feature_data and get_power) in other functions. So I can't really create a single splitter in the DefaultExtractor.extract method. I can also close this because I am right now creating a new Extractor (DefaultVMExtractor) which will include this PR and the new Extractor. @sunya-ch

Reply:
@KaiyiLiu1234 Thank you for your contribution on this PR. In my opinion, creating a new DefaultVMExtractor is more preferable.